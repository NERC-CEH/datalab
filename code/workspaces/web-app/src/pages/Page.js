import React from 'react';
import PropTypes from 'prop-types';
import { Typography, withStyles } from '@material-ui/core';
import Footer from '../components/app/Footer';

const style = theme => ({
  pageTemplate: {
    display: 'flex',
    flexDirection: 'column',
    minHeight: '100%',
  },
  contentArea: {
    flexGrow: 1,
  },
});

function Page({ title, children, classes, match }) {
  return (
    <div className={classes.pageTemplate}>
      {title ? <Typography variant="h4">{title}</Typography> : null}
      <div className={classes.contentArea}>
        {children}
      </div>
      <Footer/>
    </div>
  );
}

Page.propTypes = {
  title: PropTypes.string,
  classes: PropTypes.object.isRequired,
};

export default withStyles(style)(Page);
