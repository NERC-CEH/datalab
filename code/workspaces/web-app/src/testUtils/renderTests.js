import React from 'react';
import { render } from '@testing-library/react';
import { ThemeProvider as MuiThemeProvider } from '@mui/material/styles';
import theme from '../theme';

const AllTheProviders = ({ children }) => (
    <MuiThemeProvider theme={theme}>
        {children}
    </MuiThemeProvider>
);

const customRender = (ui, options) => render(ui, { wrapper: AllTheProviders, ...options });

export * from '@testing-library/react';
export { customRender as render };
