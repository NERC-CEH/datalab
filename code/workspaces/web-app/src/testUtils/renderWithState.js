import React from 'react';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import { render } from '@testing-library/react';
import promise from 'redux-promise-middleware';
import {
  PROMISE_TYPE_PENDING,
  PROMISE_TYPE_SUCCESS,
  PROMISE_TYPE_FAILURE,
} from '../actions/actionTypes';

const promiseTypeSuffixes = [PROMISE_TYPE_PENDING, PROMISE_TYPE_SUCCESS, PROMISE_TYPE_FAILURE];

const TEST_USER = 'test-user';

export const buildDefaultTestState = () => ({
  authentication: {
    identity: {
      sub: TEST_USER,
    },
    permissions: {
      fetching: { inProgress: false },
      value: [],
    },
  },
  projectUsers: {
    fetching: { inProgress: false },
    value: [
      { userId: TEST_USER, role: 'admin' },
    ],
  },
});

export const renderWithState = (
  initialState,
  ComponentToRender,
  props = {},
  renderOptions = {},
) => {
  const mockStore = configureStore([promise({ promiseTypeSuffixes })]);
  const store = mockStore(initialState);

  const wrapper = render(
    <Provider store={store}>
      <ComponentToRender {...props} />
    </Provider>,
    renderOptions,
  );

  return {
    wrapper,
    store,
  };
};
