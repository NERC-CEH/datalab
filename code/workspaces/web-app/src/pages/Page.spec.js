import React from 'react';
import { render } from '../testUtils/renderTests';
import Page from './Page';

jest.mock('../components/app/Footer', () => () => (<div>Footer mock</div>));

describe('Page', () => {
  it('renders correctly when title is provided', () => {
    const wrapper = render(<Page title="The Page Title">Page Content</Page>);
    expect(wrapper.container).toMatchSnapshot();
  });

  it('renders correctly when title is not provided', () => {
    const wrapper = render(<Page>Page Content</Page>);
    expect(wrapper.container).toMatchSnapshot();
  });
});
