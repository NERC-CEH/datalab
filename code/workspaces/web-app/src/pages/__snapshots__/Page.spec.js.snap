// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page renders correctly when title is not provided 1`] = `
<div>
  <div
    class="Page-pageTemplate-4"
  >
    <div
      class="Page-contentArea-5"
    >
      Page Content
    </div>
    <div>
      Footer mock
    </div>
  </div>
</div>
`;

exports[`Page renders correctly when title is provided 1`] = `
<div>
  <div
    class="Page-pageTemplate-1"
  >
    <h4
      class="MuiTypography-root MuiTypography-h4 Page-title-3 css-1yqx3hy-MuiTypography-root"
    >
      The Page Title
    </h4>
    <div
      class="Page-contentArea-2"
    >
      Page Content
    </div>
    <div>
      Footer mock
    </div>
  </div>
</div>
`;
